### PR TITLE
[WIP] Enable GUI on OSX using SDL

### PR DESCRIPTION
**Pull request by tmm1**
work in progress, but basic functionality is implemented:

screen shot 2015-07-30 at 11 55 48 pm

### CODE REVIEW

**Comment by tmm1** [comment]
Still to do:

- [ ] show help and frame info text on the screen
- [ ] make a separate `comskip-gui` binary that uses SDL_main, to avoid dock application on MacOSX when using cli only
  - [ ] make `comskip -w` re-exec `comskip-gui` automatically
- [ ] add menu bar items?

**Comment by tmm1** [comment]
The basics here are working. There's no on screen display or menu bar items, but it's still functional enough to be able to debug comskip's behavior. Merging.

**Comment by gardar** [comment]
Any plans on getting help and frame info text working on linux/mac? What's missing?

**Comment by tmm1** [comment]
It's been a couple years since I looked at this, so I don't remember. Probably just need to enable the text rendering code, and maybe point it at a font on the system?

FWIW, the help text and frame info does show up in the console window.

**Comment by gardar** [comment]
Understood, yep I noticed it in the console window. It's not a big deal but it would be nice to have comeskipper on linux on par with the windows version.

File: video_out_sdl.c [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+
+    return (vo_instance_t *) instance;
+}
+#endif
+
+void ShowHelp(char **ta)
+{
+  char *t;
+  while ((t = *ta)) {
+    printf("%s\n", t);
```
Comment (tmm1) [inline video_out_sdl.c]: @gardar Instead of this printf here (and below in `ShowDetails`), you would add code to render the text onto the SDL screen.

Comment (tmm1) [inline video_out_sdl.c]: See for example http://stackoverflow.com/questions/22886500/how-to-render-text-in-sdl2

Comment (tmm1) [inline video_out_sdl.c]: (Except that's for SDL2, so you want the SDL1 version of the equivalent code.)

**Comment by gardar** [comment]
I noticed that the debug window closes immediately after comskip runs on linux and does not wait for user interaction like the windows version, so you cannot make any changes with the gui unless you open the .txt file again.  
Should I perhaps open up a new issue for this?

**Comment by jschwalbe** [comment]
Can anyone give a How-To of sorts to get the GUI to work (either on Ubuntu or macOS)? Thanks!

**Comment by erikkaashoek** [comment]
Invoke comskip without any arguments and it will show all switches including "-w" to start the gui

**Comment by jschwalbe** [comment]
Is there any documentation on this further, or perhaps anyone else who can chime in? I'm not finding success on Ubuntu running it with -w.

**Comment by jschwalbe** [comment]
One comment: I've been running `comskip -w` and getting frustrated bc it wasn't working. Not until I built & ran `comskip-gui -w` did it work. Perhaps there should be a warning if `comskip -w` if invoked to remind user to use the gui binary? (Or just build one binary? Not sure why there's two.)

**Pull request merged**